src/main/java/com/shapesecurity/shift/fuzzer/GenCtx.java

```diff
@@ -184,6 +196,22 @@ GenCtx forbidMissingElse() {
         this.inSwitch,
         this.inStrictMode,
         this.inFunctional,
+        false,
+        this.allowReturn);
+  }
+
+  @NotNull
+  GenCtx allowReturn() {
```

Comment:
We could probably just re-use `enterFunctional`. Set `allowReturn` to `true` there, and then we don't ever need to call this.